File: clients/shared/Sources/EmailClientKit/Models/Account.swift
import Foundation

/// The type of an email account.
public enum AccountType: String, Codable, Sendable, Equatable, CaseIterable {
    case work
    case personal
}

/// The connection status of an email account.
public enum AccountStatus: String, Codable, Sendable, Equatable, CaseIterable {
    case online
    case offline
    case error
    case syncing
}

/// An email account.
public struct Account: Codable, Identifiable, Sendable, Equatable {
    public let id: String
    public let name: String
    public let emailAddress: String
    public let accountType: AccountType
    public let color: String
    public let status: AccountStatus
    public let statusMessage: String?
    public let counts: AccountCounts?

    public init(
        id: String,
        name: String,
        emailAddress: String,
        accountType: AccountType,
        color: String,
        status: AccountStatus,
        statusMessage: String? = nil,
        counts: AccountCounts? = nil
    ) {
        self.id = id
        self.name = name
        self.emailAddress = emailAddress
        self.accountType = accountType
        self.color = color
        self.status = status
        self.statusMessage = statusMessage
        self.counts = counts
    }
}

/// Email counts per view for an account.
public struct AccountCounts: Codable, Sendable, Equatable {
    public let actionQueue: Int?
    public let readingQueue: Int?
    public let filtered: Int?
    public let filteredBorderline: Int?
    public let allInboxes: Int?

    public init(
        actionQueue: Int? = nil,
        readingQueue: Int? = nil,
        filtered: Int? = nil,
        filteredBorderline: Int? = nil,
        allInboxes: Int? = nil
    ) {
        self.actionQueue = actionQueue
        self.readingQueue = readingQueue
        self.filtered = filtered
        self.filteredBorderline = filteredBorderline
        self.allInboxes = allInboxes
    }
}

File: server/internal/models/account.go
package models

import "time"

// Account type values.
const (
	AccountTypeWork     = "work"
	AccountTypePersonal = "personal"
)

// Account status values.
const (
	AccountStatusOnline  = "online"
	AccountStatusOffline = "offline"
	AccountStatusError   = "error"
	AccountStatusSyncing = "syncing"
)

// Account represents an email account.
type Account struct {
	ID            string         `json:"id"`
	Name          string         `json:"name"`
	EmailAddress  string         `json:"email_address"`
	AccountType   string         `json:"account_type"`
	Color         string         `json:"color"`
	Status        string         `json:"status"`
	StatusMessage string         `json:"status_message,omitempty"`
	Counts        *AccountCounts `json:"counts,omitempty"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
}

// AccountCounts holds per-view email counts for an account.
type AccountCounts struct {
	ActionQueue        int `json:"action_queue"`
	ReadingQueue       int `json:"reading_queue"`
	Filtered           int `json:"filtered"`
	FilteredBorderline int `json:"filtered_borderline"`
	AllInboxes         int `json:"all_inboxes"`
}

// AccountListResponse wraps the account list.
type AccountListResponse struct {
	Data []Account `json:"data"`
}
